src/legacy/core_plugins/interpreter/common/types/datatable.ts

```diff
@@ -19,75 +19,128 @@
 
 import { map, pick, zipObject } from 'lodash';
 
-export const datatable = () => ({
-  name: 'datatable',
-  validate: datatable => {
+import { ExpressionType } from '../../types';
+import { PointSeries } from './pointseries';
+import { Render } from './render';
+
+const name = 'datatable';
+
+/**
+ * A Utility function that Typescript can use to determine if an object is a Datatable.
+ * @param datatable
+ */
+export const isDatatable = (datatable: any): datatable is Datatable =>
+  !!datatable && datatable.type === 'datatable';
+
+/**
+ * This type represents the `type` of any `DatatableColumn` in a `Datatable`.
+ */
+export type DatatableColumnType = 'string' | 'number' | 'boolean' | 'date' | 'null';
+
+/**
+ * This type represents a `DatatableRow` in a `Datatable`.
+ */
+export type DatatableRow = Record<string, any>;
+
+/**
+ * This type represents the shape of a column in a `Datatable`.
+ */
+export interface DatatableColumn {
+  name: string;
+  type: DatatableColumnType;
+}
+
+/**
+ * A `Datatable` in Canvas is a unique structure that represents tabulated data.
+ */
+export interface Datatable {
+  type: typeof name;
+  columns: DatatableColumn[];
+  rows: DatatableRow[];
+}
+
+interface SerializedDatatable extends Datatable {
+  rows: string[][];
```

Comment:
I _think_ this is the correct type for a serialized datatable?

Reply:
It looks good to me, but @w33ble would probably know best.

Reply:
Confirmed this with @w33ble last week